drivers/hmis/app/models/hmis/staff_assignment.rb

```diff
@@ -0,0 +1,16 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::StaffAssignment < Hmis::HmisBase
+  acts_as_paranoid
+
+  belongs_to :data_source, class_name: 'GrdaWarehouse::DataSource'
+  belongs_to :user, class_name: 'Hmis::User'
+  belongs_to :household, class_name: 'Hmis::Hud::Household', primary_key: [:data_source_id, :HouseholdID], foreign_key: [:data_source_id, :household_id], inverse_of: :staff_assignments, optional: true
+  belongs_to :staff_assignment_type, foreign_key: :hmis_staff_assignment_type_id, class_name: 'Hmis::StaffAssignmentType'
+
+  validates :household_id, uniqueness: { scope: [:user_id, :data_source_id, :hmis_staff_assignment_type_id] }
```

Comment:
If the user tries to submit a non-unique combination of [household, user, assignment type], the error message they see is "Household has already been taken." This isn't the most intuitive error message - the problem is not specifically with household, but with the combo of the 3 fields. There's probably a better railsy way to do this, maybe a way to apply a validation that is not tied to a specific field? To discuss/investigate docs more

Reply:
remove this validation (it's already checked on the db), instead add a check to the mutation where we can customize the error message